### ISSUE TITLE
about sequence_length

### ISSUE THREAD

**Issue by xixiha5230** [MarcoMeter/recurrent-ppo-truncated-bptt#11]
This issue is mainly to thank this great project. 

I followed these codes, wrote a PPO algorithm with RNN, and trained it in the [hallway](https://github.com/Unity-Technologies/ml-agents/blob/develop/docs/Learning-Environment-Examples.md#hallway) environment of ml-agents.
I found that the parameter `sequence_length` has a great influence on the training results. I don’t know Why, I think `sequence_length` does not affect the memory length of the agent, is this correct?

The following is the reward curve, the max reward is 1: 
green line:  `sequence_length = 8` 
gray line: `sequence_length = 16` 
![image](https://user-images.githubusercontent.com/31526189/221081163-f8a86ae7-7be1-4e98-b4a7-e5b6e7c7676c.png)

**Comment by MarcoMeter** [comment]
Thanks @xixiha5230 for your interest and question.

Recurrent neural nets have an "unlimited" memory length. The major difference is that the sequence length affects backpropagation through time. Gradients, and hence optimization steps, can be larger or more meaningful because of longer sequence lengths. My impression usually is that longer sequence lengths usually perform better. Concerning Hallway, I'd try a sequence length that is closer to the length of an episode (maybe 64).

**Comment by xixiha5230** [comment]
Your answer cleared my doubts. My experiment in Hallway also shows that a longer `sequence_length` has a better effect. Thank you again, and I wish you all the best.
